Detections/Syslog/ssh_NewlyInternetExposed.yaml

```diff
@@ -0,0 +1,57 @@
+﻿id: 4915c713-ab38-432e-800b-8e2d46933de6
+name: ssh newly internet-exposed endpoints
+description: endpoints with a history of sign-ins only from private IP addresses are accessed from a public IP address.
+severity: Medium
+requiredDataConnectors:
+  - connectorId: Syslog
+    dataTypes:
+      - Syslog
+queryFrequency: 24h
+queryPeriod: 7d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1078,T1110
+query: |
+
+let PrivateIPregex = @'^127\.|^10\.|^172\.1[6-9]\.|^172\.2[0-9]\.|^172\.3[0-1]\.|^192\.168\.'; 
+let ssh_logins = Syslog
+  | where TimeGenerated >= ago(7d)
+  | where Facility contains "auth" and ProcessName != "sudo"
+  | where SyslogMessage has "Accepted"
+  | extend SourceIP = extract("(([0-9]|[0-9][0-9]|[0-9][0-9][0-9])\\.([0-9]|[0-9][0-9]|[0-9][0-9][0-9])\\.([0-9]|[0-9][0-9]|[0-9][0-9][0-9])\\.([0-9]|[0-9][0-9]|[0-9][0-9][0-9]))",1,SyslogMessage) 
```

Comment:
If you want to make things a bit more readable you can adjust this regex to:
(([0-9]{1,3})\\.([0-9]{1,3})\\.([0-9]{1,3})\\.(([0-9]{1,3})))

I've used the expanded one in the past due to issues with { in queries defined in Notebooks, but I think in these queries this format is a bit neater.

Reply:
done